lib/std/outplaces.nim

```diff
@@ -0,0 +1,46 @@
+import std/macros
+include system/inclrtl
+
+proc outplaceImpl(arg, call: NimNode): NimNode =
+  expectKind call, nnkCallKinds
+  let tmp = genSym(nskVar, "outplaceResult")
+  var callsons = call[0..^1]
+  var found = false
+  for i in 1..<len(callsons):
+    if callsons[i].kind == nnkIdent and callsons[i].strVal == "_":
+      callsons[i] = tmp
+      found = true
+      break
+  if not found: callsons.insert(tmp, 1)
+  result = newTree(nnkStmtListExpr,
+    newVarStmt(tmp, arg),
+    copyNimNode(call).add callsons,
+    tmp)
+
+proc replaceOutplace(lhs, n: NimNode): NimNode =
+  case n.kind
+  of nnkDotExpr, nnkBracketExpr:
+    result = copyNimTree(n)
+    result[0] = replaceOutplace(lhs, result[0])
+  of nnkCall:
+    result = outplaceImpl(lhs, n)
+  of nnkCommand:
+    result = outplaceImpl(lhs, n)
+  else:
+    doAssert false, "unexpected kind: " & $n.kind
+
+macro `.>`*(lhs, rhs): untyped {.since: (1, 1).} =
+  ## Outplace operator: turns an `in-place`:idx: algorithm into one that works on
+  ## a copy and returns this copy. A placeholder `_` can optionally be used to
+  ## specify an output parameter of position > 0.
+  ## **Since**: Version 1.2.
```

Comment:
Paragraphs! Please add empty line above this, otherwise it'll render poorly in the doc gen.

Also, you need to add this file to the docgen config IIRC.

Reply:
> Please add empty line above this

done

> Also, you need to add this file to the docgen config IIRC.

I believe I've fixed this in https://github.com/nim-lang/Nim/pull/13221